apple/internal/rule_factory.bzl

```diff
@@ -170,6 +163,23 @@ def _create_apple_test_rule(*, doc, implementation, platform_type):
         # The aspect is withheld to avoid unnecessary overhead in this instance of `test_host`, and
         # the provider is unnecessarily generic to accomodate any possible value of `test_host`.
         rule_attrs.test_host_attrs(aspects = [], providers = [[AppleBundleInfo]]),
+        # Add bundle attributes to make them visible in documentation
+        rule_attrs.infoplist_attrs(),
+        {
+            "bundle_name": attr.string(
+                mandatory = False,
+                doc = """
+The desired name of the bundle (without the extension). If this attribute is not set, then the name
+of the target will be used instead.
+""",
+            ),
+            "frameworks": attr.label_list(
+                providers = [[AppleBundleInfo]],
+                doc = """
+A list of framework targets that this target depends on.
+""",
+            ),
+        },
```

Comment:
This causes the test runner to be invalidated when these values change. I don't like that. Longer term it might be best to separate the test runner from the bundle (https://github.com/bazelbuild/rules_apple/pull/2744#issuecomment-3021218046).

Short term, we could adjust `apple/ios.doc.bzl` and the like to point to the macro in `apple/ios.bzl` and add a hand-crafted docstring.